src/webapi/xml_http_request.rs

```diff
@@ -0,0 +1,134 @@
+use webapi::event_target::{IEventTarget, EventTarget};
+use webcore::unsafe_typed_array::UnsafeTypedArray;
+use webcore::value::{
+    Reference,
+    Value,
+};
+use webcore::try_from::TryInto;
+
+/// Use XMLHttpRequest (XHR) objects to interact with servers.
+/// You can retrieve data from a URL without having to do a full page refresh.
+/// This enables a Web page to update just part of a page without disrupting
+/// what the user is doing. XMLHttpRequest is used heavily in Ajax programming.
+/// 
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest)
+pub struct XMLHttpRequest( Reference );
+
+reference_boilerplate! {
+    XMLHttpRequest,
+    instanceof XMLHttpRequest
+    convertible to EventTarget
+}
+
+/// An enum indicating the state of the `XMLHttpRequest`.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState)
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum ReadyState {
+    /// Client has been created. [open()](struct.XMLHttpRequest.html#method.open) not called yet.
+    Unsent,
+    /// [open()](struct.XMLHttpRequest.html#method.open) has been called.
+    Opened,
+    /// [send()](struct.XMLHttpRequest.html#method.send) has been called, and headers and [status()](struct.XMLHttpRequest.html#method.status) are available.
+    HeadersReceived,
+    /// Downloading; [reponse_text()](struct.XMLHttpRequest.html#method.reponse_text) holds partial data.
+    Loading,
+    /// The operation is complete.
+    Done,
+}
+
+impl IEventTarget for XMLHttpRequest {}
+
+
+impl XMLHttpRequest {
+    /// Creates new `XMLHttpRequest`.
+    pub fn new() -> XMLHttpRequest {
+        XMLHttpRequest(js!{
```

Comment:
Actually, you should be able to convert it like this:

```
js!( return new XMLHttpRequest(); ).try_into().unwrap()
```

Reply:
I think I tried that originally and I couldn't get it to work. I wonder if I set something up wrong with the boilerplate macro. I'll try it again tonight, maybe I just made a silly mistake.

Reply:
I must have messed up something silly cause it worked fine this time. Thanks.